### PR TITLE
Added back running of spark core modules tests

### PR DESCRIPTION
**Pull request by chtyim**
Also added two new profiles for spark development in IDE
  - Use symlink instead of copy source so that there is single place to modify the common code.

### CODE REVIEW

**Comment by chtyim** [comment]
Build: https://builds.cask.co/browse/CDAP-DUT5730

File: cdap-spark-core/pom.xml [hunk 1]
```diff
@@ -226,13 +226,99 @@
...
           </execution>
+          <execution>
+            <id>add-test-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>add-test-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${project.build.directory}/generated-sources/cdap-spark-core-base/src/test/java</source>
```
Comment (albertshau) [inline cdap-spark-core/pom.xml]: should be generated-test-sources

File: cdap-spark-core/pom.xml [hunk 2]
```diff
@@ -226,13 +226,99 @@
...
+              </execution>
+              <execution>
+                <id>symlink-test-source</id>
+                <phase>generate-sources</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <target>
+                    <symlink link="${project.build.directory}/generated-sources/cdap-spark-core-base/src/test" resource="${spark.base.dir}/src/test" overwrite="true" />
```
Comment (albertshau) [inline cdap-spark-core/pom.xml]: should be generated-test-sources

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by chtyim** [comment]
Build passed.

**Pull request merged**